content/articles/how-to-build-a-vue-app-with-flask-sqlite-backend-using-docker/index.md

```diff
@@ -0,0 +1,952 @@
+    app.run(debug=True)
+```
+
+From above, we are importing the Flask module, initializing it, and starting it.
+
+#### Setting up the database
+To set up the database, follow the following steps:
+
+- Import the following packages:
+
+```python
+from flask_sqlalchemy import SQLAlchemy
+from flask_marshmallow import Marshmallow
+import os
+```
+
+- Set up the base directory for the application:
+
+```python
+basedir = os.path.abspath(os.path.dirname(__file__))
+```
+
+- Set up the database application configuration
+
+```python
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///' + os.path.join(basedir,'db.sqlite')
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+```
+
+- Initialize the database
+
+```python
+db = SQLAlchemy(app)
+```
+
+- Initialize marshmallow
+
+```python
+ma = Marshmallow(app)
+```
+
+- Set up the todo model. It will make up a sample todo stored in the database.
+
+```python
+class Todo(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String(100))
+    description = db.Column(db.String(400))
+
+    def __init__(self,title,description):
+        # Add the data to the instance
+        self.title = title
+        self.description = description
+```
+
+From above, we are defining that a Todo will have an id, title, and description.
+
+- Set up the todo schema. It will be called when querying the todos data.
+
+```python
+class TodoSchema(ma.Schema):
+    class Meta:
+        fields = ('id','title','description')
+```
+
+From above, we define that for every todo, we will be interested in the id, title, and description.
+
+#### Initialize the schema
+To initialize the schema, we have to do it differently for a single todo and for multiple todos. For this, we will add the following:
+
+```python
+todo_schema = TodoSchema()
+todos_schema = TodoSchema(many=True)
+```
+
+The first is for a single todo; the other is for multiple todos.
+
+#### Setting up routes
+To set up the routes,
+
+- Start by importing packages:
+
+```python
+from flask import Flask,request, jsonify
+from flask_cors import CORS,cross_origin
+```
+
+`request` will be used to get the payload (data sent), whereas `jsonify` will be used to return JSON data. `CORS` and `cross_origin` for setting up the access policy.
+
+- Cors configuration for the entire app:
+
+```python
+CORS(app,resources={r"/api": {"origins": "*"}})
+app.config['CORS_HEADERS'] = 'Content-Type'
+```
+
+From above, we accept all origins hitting the `/api` endpoint from which we will expose the API.
+
+- Creating a todo: the following route creates a todo:
+
+```python
+@app.route('/api/todo', methods=['POST'])
+@cross_origin(origin='*',headers=['content-type'])
+def add_todo():
+# get the data
+    title = request.json['title']
+    description = request.json['description']
+
+    # Create an instance
+    new_todo = Todo(title, description)
+
+    # Save the todo in the db
+    db.session.add(new_todo)
+    db.session.commit()
+
```

Comment:
You have not created the tables in the database yet. ie. `db.create_all()`. First create the tables for the app to work.

Reply:
The app is using marshmallow to auto-generate the schemas